sno/upgrade/__init__.py

```diff
@@ -1,15 +1,39 @@
 import click
+import pygit2
 
-from . import upgrade_00_02, upgrade_02_05  # noqa
-from sno.cli_util import add_help_subcommand
+from pathlib import Path
 
+from . import upgrade_v0, upgrade_v1
+from sno.exceptions import InvalidOperation
+from sno.repository_version import get_repo_version
 
-@add_help_subcommand
-@click.group()
-def upgrade():
-    """ Upgrade repositories between versions of Sno """
-    pass
 
+@click.command()
+@click.pass_context
+@click.argument("source", type=click.Path(exists=True, file_okay=False), required=True)
+@click.argument("dest", type=click.Path(exists=False, writable=True), required=True)
+def upgrade(ctx, source, dest):
+    """
+    Upgrade a repository for an earlier version of Sno to be compatible with the latest version.
+    The current repository structure of Sno is known as Datasets V2, which is used from Sno 0.5 onwards.
+    """
+    source = Path(source)
+    dest = Path(dest)
 
-upgrade.add_command(upgrade_00_02.upgrade, name='00-02')
-upgrade.add_command(upgrade_02_05.upgrade, name='02-05')
+    if dest.exists():
+        raise click.BadParameter(f"'{dest}': already exists", param_hint="DEST")
+
+    source_repo = pygit2.Repository(str(source))
+    if not source_repo or not source_repo.is_bare:
+        raise click.BadParameter(
+            f"'{source}': not an existing repository", param_hint="SOURCE"
+        )
+
+    version = get_repo_version(source_repo)
+
+    if version == 0:
+        upgrade_v0.upgrade_v0(ctx, source, dest)
```

Comment:
This upgrades it from v0 to v1, right? Shouldn't it then call upgrade_v1 to go the rest of the way?

Reply:
Nup I changed it to upgrade straight to V2.